tests/cuda/test_cuda_allocators.cpp

```diff
@@ -41,52 +39,17 @@ void run_host_tests() {
    return;
 }
 
-/// Function running simple tests with the active memory manager.
-void run_device_tests() {
-
-   // Create the test vector.
-   test_vector< int > testv;
-
-   // Resize it in a few different ways.
-   testv.resize( 100 );
-   testv.reserve( 1000 );
-   testv.resize( 10 );
-   return;
-}
-
 int main() {
+   vecmem::cuda::host_memory_resource host_resource;
+   vecmem::cuda::managed_memory_resource managed_resource;
 
    // Run the tests with all available memory managers, that allow for access to
    // the memory from the host.
-   vecmem::memory_manager::instance().set(
-      std::make_unique< vecmem::cuda::arena_memory_manager >(
-         vecmem::cuda::arena_memory_manager::memory_type::host ) );
-   run_host_tests();
-   vecmem::memory_manager::instance().set(
-      std::make_unique< vecmem::cuda::arena_memory_manager >(
-         vecmem::cuda::arena_memory_manager::memory_type::managed ) );
-   run_host_tests();
-
-   vecmem::memory_manager::instance().set(
-      std::make_unique< vecmem::cuda::direct_memory_manager >(
-         vecmem::cuda::direct_memory_manager::memory_type::host ) );
-   run_host_tests();
-   vecmem::memory_manager::instance().set(
-      std::make_unique< vecmem::cuda::direct_memory_manager >(
-         vecmem::cuda::direct_memory_manager::memory_type::managed ) );
-   run_host_tests();
-
-   // Run much simpler tests with the memory managers that only allocate memory
-   // on the device(s).
-   vecmem::memory_manager::instance().set(
-      std::make_unique< vecmem::cuda::arena_memory_manager >(
-         vecmem::cuda::arena_memory_manager::memory_type::device ) );
-   run_device_tests();
+   run_host_tests(host_resource);
+   run_host_tests(managed_resource);
 
-   vecmem::memory_manager::instance().set(
-      std::make_unique< vecmem::cuda::direct_memory_manager >(
-         vecmem::cuda::direct_memory_manager::memory_type::device ) );
-   run_device_tests();
+   run_host_tests(host_resource);
+   run_host_tests(managed_resource);
```

Comment:
I believe these two are not needed here.

Reply:
Yes, the reason they're still here is that something similar was being done in the old tests as well. I figured that might have been there to check if everything still works after switching back to host allocations? But I'm more than happy to remove these calls.

Reply:
Previously the test used 2 different "memory managers". Once we'll have multiple "memory resources" (built on top of the basic ones), we can add back the extra tests. But for now just testing `host_resource` and `managed_resource` once should be enough. :wink: